pkg/pubsub/client/pubsub_client.go

```diff
@@ -0,0 +1,80 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+)
+
+func init() {
+	flag.Var(pubSubs, "pub-sub-provider", "Preferred pub-sub provider, e.g dapr, rabbitmq. This flag can be declared more than once. Omitting will default to not using any pub sub provider.")
```

Comment:
This flag should list the valid values that have been compiled in (keys of `supportedPubSub`)

Also, it's not a "preferred" pubsub provider if it can be provided more than once. "Enabled" pubsub providers?

Finally, is there a reason this flag needs to exist? What use case are we satisfying by including it?

Reply:
Later down the path where there exist more drivers, this flag will be useful for users to enable providers right?

Reply:
If we just had all providers enabled always (i.e. no flag), wouldn't the existence/absence of a config referencing a particular provider determine whether that provider was used?

The strongest argument I can think of for a flag would be if admins wanted to lock down which providers are permissible to use, though that's fairly speculative, since I'd assume provider configs would be a fairly trusted operation.

IMO I'd leave it out of the MVP and see how things develop.

Reply:
Makes sense, I removed `--pub-sub-provider` flag. Added `--enable-pub-sub` to enable/disable config controller for pubsub feature.